src/crystal/system/unix/path.cr

```diff
@@ -9,15 +9,15 @@ module Crystal::System::Path
 
       pwd = uninitialized LibC::Passwd
       pwd_pointer = pointerof(pwd)
-      ret = uninitialized LibC::Int
+      ret = nil
       System.retry_with_buffer("getpwuid_r", User::GETPW_R_SIZE_MAX) do |buf|
         ret = LibC.getpwuid_r(id, pwd_pointer, buf, buf.size, pointerof(pwd_pointer))
       end
 
       if pwd_pointer
         String.new(pwd.pw_dir)
       else
-        raise RuntimeError.from_os_error("getpwuid_r", Errno.new(ret))
+        raise RuntimeError.from_os_error("getpwuid_r", Errno.new(ret.not_nil!))
```

Comment:
I think this is not exactly using the API correctly. `getpwuid_r` returns `0` and sets the result to a null pointer if not matching password record was found. That would result in showing a runtime error  "success".
We should handle the case `ret.zero?` and `pwd_pointer.nil?` to generate an appropriate error message.

*Most* actual error codes are already handled by `retry_with_buffer`. I'm not sure if `getpwuid_r` can even return any of the ones that are ignored (and make it return `nil`): `ENOENT`, `ESRCH`, `EBADF`, `EPERM`. But having a generic fallback in cause anyone of these happens doesn't hurt.

Reply:
What is a situation where `getpwuid(getuid())` fails because no matching UID is found?

Reply:
I understand that would be `ret.zero? && pwd_pointer.nil?`.

Reply:
That is not what I am asking. If I pass an arbitrary user ID, then it can happen that no matching record is found. What is a situation where the _current real user ID_ has no matching record?

Reply:
I don't know.